src/axom/multimat/multimat.cpp

```diff
@@ -260,58 +425,80 @@ void MultiMat::setCellMatRel(vector<bool>& vecarr, DataLayout layout)
   relDenseSet(layout) = ProductSetType(&set1, &set2);
 
   //Create a field for VolFrac as the 0th field
-  m_mapVec.emplace_back(nullptr);
   m_fieldNameVec.push_back("Volfrac");
+  m_fieldBackingVec.emplace_back(new FieldBacking);
   m_fieldMappingVec.push_back(FieldMapping::PER_CELL_MAT);
   m_dataTypeVec.push_back(DataTypeSupported::TypeDouble);
   m_fieldDataLayoutVec.push_back(DataLayout::CELL_DOM);
   m_fieldSparsityLayoutVec.push_back(SparsityLayout::SPARSE);
+  m_fieldStrideVec.push_back(1);
 
-  SLIC_ASSERT(m_mapVec.size() == 1);
   SLIC_ASSERT(m_fieldNameVec.size() == 1);
   SLIC_ASSERT(m_fieldMappingVec.size() == 1);
   SLIC_ASSERT(m_dataTypeVec.size() == 1);
   SLIC_ASSERT(m_fieldDataLayoutVec.size() == 1);
   SLIC_ASSERT(m_fieldSparsityLayoutVec.size() == 1);
+  SLIC_ASSERT(m_fieldStrideVec.size() == 1);
+}
+
+void MultiMat::removeField(const std::string& field_name)
+{
+  int field_index = getFieldIdx(field_name);
+
+  if(field_index == 0)
+  {
+    // This is the volume fractions array. Issue a warning to the user.
+    SLIC_WARNING("Multimat Error: cannot remove volume fractions array.");
+  }
+  else if(field_index < 0)
+  {
+    // Field does not exist.
+    SLIC_WARNING("Multimat Error: field with name \"" << field_name
+                                                      << "\" does not exist.");
+  }
+  else
+  {
+    m_fieldNameVec.erase(m_fieldNameVec.begin() + field_index);
```

Comment:
Would it make sense to use `sidre::IndexedCollection`s for these?
That is what `sidre::Group`s use for their underlying indexing of  children `View`s and `Group`s.
The advantage would be that it tracks indices and free spaces for us.


Perhaps we can move that class to core if we do not want `multimat` to depend on `sidre`.

If we want to do this, it should be done in a follow-up PR.